common/src/sql/dbinit.sql

```diff
@@ -861,8 +865,9 @@ CREATE UNIQUE INDEX ON omicron.public.instance (
 
 -- Allow looking up instances by server. This is particularly
 -- useful for resource accounting within a sled.
-CREATE INDEX ON omicron.public.instance (
-    active_sled_id
+CREATE UNIQUE INDEX ON omicron.public.instance (
```

Comment:
This is one of the cases mentioned in the PR description. To paginate the instances on a sled, the id needs to be here, and then this can be a UNIQUE index.